CleverTapSDK/EventDatabase/CTEventDatabase.m

```diff
@@ -0,0 +1,564 @@
+                     deviceID:(NSString *)deviceID {
+    if (!_eventDatabase) {
+        CleverTapLogInternal(self.config.logLevel, @"%@ Event database is not open, cannot execute SQL.", self);
+        return 0;
+    }
+
+    const char *querySQL = "SELECT lastTs FROM CTUserEventLogs WHERE normalizedEventName = ? AND deviceID = ?";
+    __block NSInteger lastTs = 0;
+
+    dispatch_sync(_databaseQueue, ^{
+        sqlite3_stmt *statement;
+        if (sqlite3_prepare_v2(_eventDatabase, querySQL, -1, &statement, NULL) == SQLITE_OK) {
+            sqlite3_bind_text(statement, 1, [normalizedEventName UTF8String], -1, SQLITE_TRANSIENT);
+            sqlite3_bind_text(statement, 2, [deviceID UTF8String], -1, SQLITE_TRANSIENT);
+            
+            if (sqlite3_step(statement) == SQLITE_ROW) {
+                lastTs = sqlite3_column_int(statement, 0);
+            } else {
+                CleverTapLogInternal(self.config.logLevel, @"%@ No event found with eventName: %@ and deviceID: %@", self, normalizedEventName, deviceID);
+            }
+            
+            sqlite3_finalize(statement);
+        } else {
+            CleverTapLogInternal(self.config.logLevel, @"%@ SQL prepare query error: %s", self, sqlite3_errmsg(_eventDatabase));
+        }
+    });
+
+    return lastTs;
+}
+
+- (CleverTapEventDetail *)getEventDetail:(NSString *)normalizedEventName
+                                deviceID:(NSString *)deviceID {
+    if (!_eventDatabase) {
+        CleverTapLogInternal(self.config.logLevel, @"%@ Event database is not open, cannot execute SQL.", self);
+        return nil;
+    }
+
+    const char *querySQL = "SELECT eventName, normalizedEventName, count, firstTs, lastTs, deviceID FROM CTUserEventLogs WHERE normalizedEventName = ? AND deviceID = ?";
+    __block CleverTapEventDetail *eventDetail = nil;
+    
+    dispatch_sync(_databaseQueue, ^{
+        sqlite3_stmt *statement;
+        if (sqlite3_prepare_v2(_eventDatabase, querySQL, -1, &statement, NULL) == SQLITE_OK) {
+            sqlite3_bind_text(statement, 1, [normalizedEventName UTF8String], -1, SQLITE_TRANSIENT);
+            sqlite3_bind_text(statement, 2, [deviceID UTF8String], -1, SQLITE_TRANSIENT);
+            
+            if (sqlite3_step(statement) == SQLITE_ROW) {
+                const char *eventName = (const char *)sqlite3_column_text(statement, 0);
+                const char *normalizedEventName = (const char *)sqlite3_column_text(statement, 1);
+                NSInteger count = sqlite3_column_int(statement, 2);
+                NSInteger firstTs = sqlite3_column_int(statement, 3);
+                NSInteger lastTs = sqlite3_column_int(statement, 4);
+                const char *deviceID = (const char *)sqlite3_column_text(statement, 5);
+                
+                eventDetail = [[CleverTapEventDetail alloc] init];
+                eventDetail.eventName = [NSString stringWithUTF8String:eventName];
```

Comment:
Check all strings if they are not nil otherwise it will throw an exception when trying to create the NSString.

`eventDetail.eventName = eventName ? [NSString stringWithUTF8String:eventName] : nil;`

Do this for all strings.